server/src/com/thoughtworks/go/server/service/UserService.java

```diff
@@ -258,6 +258,39 @@ public void deleteUser(String username, HttpLocalizedOperationResult result) {
         }
     }
 
+    public void deleteUsers(List<String> userNames, HttpLocalizedOperationResult result) {
+        if (userNames == null || userNames.isEmpty()) {
+            result.notFound(LocalizedMessage.string("NO_USERS_SELECTED"), HealthStateType.notFound());
+            return;
+        }
+
+        Set<String> disabledUsers = allDisabledUserNames();
+
+        for (String username : userNames) {
+            if(!disabledUsers.contains(username)){
+                result.notFound(LocalizedMessage.string("USER_DISABLED_OR_NOT_FOUND", "users", userNames), HealthStateType.notFound());
+                return;
+            }
+        }
+        for (String username : userNames) {
+            deleteUser(username, result);
```

Comment:
This still does not guarantee atomicity. Can you make API calls to check the case?